fili-core/src/test/groovy/com/yahoo/bard/webservice/async/AsyncInvalidApiRequest.groovy

```diff
@@ -0,0 +1,44 @@
+package com.yahoo.bard.webservice.async
+
+import com.yahoo.bard.webservice.util.GroovyTestUtils
+
+/**
+ * Verifies that when the user has a syntax error (or some other error that triggers an InvalidApiRequestException) that
+ * the error is returned immediately.
+ */
+class AsyncInvalidApiRequest extends AsyncFunctionalSpec {
+
+    String EXPECTED_ERROR_MESSAGE =
+            """{
+                    "status": 400,
+                    "statusName": "Bad Request",
+                    "reason": "com.yahoo.bard.webservice.web.BadApiRequestException",
+                    "description": "Required parameter metrics is missing or empty. Use 'metrics=METRICNAME1,METRICNAME2' in the query string.",
+                    "druidQuery": null
+            }"""
+
+    @Override
+    Map<String, Closure<String>> getResultsToTargetFunctions() {
+        [ data: { "data/shapes/day" } ]
+    }
+
+    @Override
+    Map<String, Closure<Void>> getResultAssertions() {
+        [ data: { assert GroovyTestUtils.compareJson(it.readEntity(String), EXPECTED_ERROR_MESSAGE) } ]
+    }
+
+    @Override
+    Map<String, Closure<Map<String, List<String>>>> getQueryParameters() {
+        return [
+                data: { [
+                        asyncAfter: ["0"],
+                        dateTime: ["2016-08-30/2016-08-31"]
+                ] }
+        ]
+    }
+
+    @Override
+    Closure<String> getFakeDruidResponse() {
+        return {"I will never leave this curly prison."}
```

Comment:
Personally, I think that depends on how often we expect to have tests that ignore the results from Druid. I don't like the idea of an exceptional case being the default. Every test that I've written except for this one do interact with Druid, so it's exceptional in this case, at least.

Now that I'm done using fancy speak that makes me sound like a craftsman, the real reason is that I'm lazy, and one nice thing about making it abstract is that with a few keystrokes in IntelliJ I can automatically generate method signatures for all the abstract methods. If I make this non-abstract, I can generate all the methods but one that I need to override, and then need to generate the `getFakeDruidResponse` method signature as a separate step. 

Not a big deal, but I find it mildly annoying when overriding the `BaseDataServletComponentSpec`, and I figured that faking Druid responses would be common enough to justify making it abstract to save myself a few keystrokes.